src/main/java/hudson/matrix/MatrixConfiguration.java

```diff
@@ -492,17 +491,34 @@ public boolean scheduleBuild(List<? extends Action> actions, Cause c) {
      */
     public static class ParentBuildAction extends InvisibleAction implements QueueAction {
         
+        /**
+         * @deprecated use {@link #getMatrixBuild()} instead.  
+         */
+        @Deprecated
         public transient MatrixBuild parent;
+        private String parentId;
 
         public ParentBuildAction() {
             final Executor currentExecutor = Executor.currentExecutor();
             this.parent = currentExecutor != null 
                     ? (MatrixBuild)currentExecutor.getCurrentExecutable() : null;
+            parentId = parent != null ? parent.getExternalizableId() : null;
         }
         
         public boolean shouldSchedule(List<Action> actions) {
             return true;
         }
+        
+        public MatrixBuild getMatrixBuild() {
```

Comment:
@oleg-nenashev I am not experienced with with java and jenkins. What annotations would make sense?